go/arrow/compute/exprs/exec.go

```diff
@@ -0,0 +1,631 @@
+					if err != nil {
+						return compute.ExecBatch{}, err
+					}
+				}
+				out.Values[i] = compute.NewDatum(val)
+			}
+			return
+		}
+	}
+
+	return out, fmt.Errorf("%w: MakeExecBatch from %s", arrow.ErrNotImplemented, partial)
+}
+
+// ToArrowSchema takes a substrait NamedStruct and an extension set (for
+// type resolution mapping) and creates the equivalent Arrow Schema.
+func ToArrowSchema(base types.NamedStruct, ext ExtensionIDSet) (*arrow.Schema, error) {
+	fields := make([]arrow.Field, len(base.Names))
+	for i, typ := range base.Struct.Types {
+		dt, nullable, err := FromSubstraitType(typ, ext)
+		if err != nil {
+			return nil, err
+		}
+		fields[i] = arrow.Field{
+			Name:     base.Names[i],
+			Type:     dt,
+			Nullable: nullable,
+		}
+	}
+
+	return arrow.NewSchema(fields, nil), nil
+}
+
+type (
+	regCtxKey struct{}
+	extCtxKey struct{}
+)
+
+func WithExtensionRegistry(ctx context.Context, reg *ExtensionIDRegistry) context.Context {
+	return context.WithValue(ctx, regCtxKey{}, reg)
+}
+
+func GetExtensionRegistry(ctx context.Context) *ExtensionIDRegistry {
+	v, ok := ctx.Value(regCtxKey{}).(*ExtensionIDRegistry)
+	if !ok {
+		v = DefaultExtensionIDRegistry
+	}
+	return v
+}
+
+func WithExtensionIDSet(ctx context.Context, ext ExtensionIDSet) context.Context {
+	return context.WithValue(ctx, extCtxKey{}, ext)
+}
+
+func GetExtensionIDSet(ctx context.Context) ExtensionIDSet {
+	v, ok := ctx.Value(extCtxKey{}).(ExtensionIDSet)
+	if !ok {
+		return NewExtensionSet(
+			expr.NewEmptyExtensionRegistry(&extensions.DefaultCollection),
+			GetExtensionRegistry(ctx))
+	}
+	return v
+}
+
+func literalToDatum(mem memory.Allocator, lit expr.Literal, ext ExtensionIDSet) (compute.Datum, error) {
+	switch v := lit.(type) {
+	case *expr.PrimitiveLiteral[bool]:
+		return compute.NewDatum(scalar.NewBooleanScalar(v.Value)), nil
+	case *expr.PrimitiveLiteral[int8]:
+		return compute.NewDatum(scalar.NewInt8Scalar(v.Value)), nil
+	case *expr.PrimitiveLiteral[int16]:
+		return compute.NewDatum(scalar.NewInt16Scalar(v.Value)), nil
+	case *expr.PrimitiveLiteral[int32]:
+		return compute.NewDatum(scalar.NewInt32Scalar(v.Value)), nil
+	case *expr.PrimitiveLiteral[int64]:
+		return compute.NewDatum(scalar.NewInt64Scalar(v.Value)), nil
+	case *expr.PrimitiveLiteral[float32]:
+		return compute.NewDatum(scalar.NewFloat32Scalar(v.Value)), nil
+	case *expr.PrimitiveLiteral[float64]:
+		return compute.NewDatum(scalar.NewFloat64Scalar(v.Value)), nil
+	case *expr.PrimitiveLiteral[string]:
+		return compute.NewDatum(scalar.NewStringScalar(v.Value)), nil
+	case *expr.PrimitiveLiteral[types.Timestamp]:
+		return compute.NewDatum(scalar.NewTimestampScalar(arrow.Timestamp(v.Value), &arrow.TimestampType{Unit: arrow.Microsecond})), nil
+	case *expr.PrimitiveLiteral[types.TimestampTz]:
+		return compute.NewDatum(scalar.NewTimestampScalar(arrow.Timestamp(v.Value),
+			&arrow.TimestampType{Unit: arrow.Microsecond, TimeZone: TimestampTzTimezone})), nil
```

Comment:
What is TimestampTzTimezone?

Reply:
It's a constant defined as "UTC", defined at the top of `types.go`, matching the inline function `TimestampTzTimezoneString` in the c++ (`cpp/src/arrow/engine/substrait/type_internal.h`)